autoconsent/autoconsent-impl/src/main/java/com/duckduckgo/autoconsent/impl/store/AutoconsentSettingsDataStore.kt

```diff
@@ -27,25 +31,32 @@ interface AutoconsentSettingsDataStore {
 
 class RealAutoconsentSettingsDataStore constructor(
     private val context: Context,
-    onByDefault: Boolean,
-) :
-    AutoconsentSettingsDataStore {
+    private val autoconsentFeature: AutoconsentFeature,
+    private val appCoroutineScope: CoroutineScope,
+    private val dispatcherProvider: DispatcherProvider,
+) : AutoconsentSettingsDataStore {
 
     private val preferences: SharedPreferences by lazy { context.getSharedPreferences(FILENAME, Context.MODE_PRIVATE) }
-    private var internalUserSettings: Boolean
+    private var cachedInternalUserSetting: Boolean? = null
+    private val defaultValue: Boolean by lazy { autoconsentFeature.onByDefault().isEnabled() }
 
     init {
-        internalUserSettings = preferences.getBoolean(AUTOCONSENT_USER_SETTING, onByDefault)
+        appCoroutineScope.launch(dispatcherProvider.io()) {
+            cachedInternalUserSetting = preferences.getBoolean(AUTOCONSENT_USER_SETTING, defaultValue)
+        }
     }
 
     override var userSetting: Boolean
```

Comment:
I think this should be a suspend fun instead

Reply:
Probably, but I didn't want to introduce big changes here since this is accessed from multiple places.